cpu/sam0_common/periph/uart.c

```diff
@@ -389,24 +416,25 @@ static inline void irq_handler_tx(unsigned uartnum)
 static inline void irq_handler(unsigned uartnum)
 {
     uint32_t status = dev(uartnum)->INTFLAG.reg;
+    dev(uartnum)->INTFLAG.reg = status;
 
 #if !defined(UART_HAS_TX_ISR) && defined(MODULE_PERIPH_UART_NONBLOCKING)
     if ((status & SERCOM_USART_INTFLAG_DRE) && dev(uartnum)->INTENSET.bit.DRE) {
         irq_handler_tx(uartnum);
     }
 #endif
 
+#ifdef MODULE_PERIPH_UART_RXSTART_IRQ
+    if (status & SERCOM_USART_INTFLAG_RXS && dev(uartnum)->INTENSET.bit.RXS) {
+        uart_ctx[uartnum].rxs_cb(uart_ctx[uartnum].rxs_arg);
+    }
+#endif
+
     if (status & SERCOM_USART_INTFLAG_RXC) {
         /* interrupt flag is cleared by reading the data register */
         uart_ctx[uartnum].rx_cb(uart_ctx[uartnum].arg,
                                 (uint8_t)(dev(uartnum)->DATA.reg));
     }
-#ifdef SERCOM_USART_INTFLAG_ERROR
-    else if (status & SERCOM_USART_INTFLAG_ERROR) {
-        /* clear error flag */
-        dev(uartnum)->INTFLAG.reg = SERCOM_USART_INTFLAG_ERROR;
-    }
-#endif
```

Comment:
Unrelated change?

Reply:
Ah I see. You are clearing pending IRQs in the first lines of the ISR ... sorry I haven't seen this.